index.bs

```diff
@@ -967,6 +970,39 @@ A string or dictionary identifying a cryptographic algorithm and optionally a se
 defined in [[!WebCryptoAPI]].
 
 
+## Algorithm to Relax the Same-Origin Restriction ## {#algo-relax-same-origin-restriction}
+
+In general, the same-origin policy prevents web applications from interacting
+within user agents, unless they have the same origin. However, there are many
+specific facets and exceptions to this overall policy.
+
+For example, cookies [[RFC6265]] have same-origin restrictions similar to those
+of the HTML `document.domain` attribute. Both permit <a>relaxing the same-origin
+restriction</a> in a particular fashion, under certain circumstances.
+
+The below algorithm implements a same-origin relaxation in the same vein as
+cookies and `document.domain`, but without modifying the <a>Document</a> object.
+
+1. If the given value is the empty string, then throw a "{{SecurityError}}" <a>DOMException</a>.
+2. Let |host| be the result of <a link-for='url'>parsing</a> the given value.
+3. If |host| is failure, then throw a "{{SecurityError}}" <a>DOMException</a>.
+4. If this <a>Document</a> object’s <a link-for='web'>origin</a> is an <a link-for='web'>opaque origin</a>, then throw a "{{SecurityError}}" <a>DOMException</a>.
```

Comment:
Which "this Document"?  There is no Document passed to this algorithm, and the caller doesn't make it clear which Document is the one involved (current, incumbent, relevant, entry, something else?).

I suggest having this algorithm explicitly take two arguments: a Document and a string.  And then having callers explicitly say what they pass to it.